src/hotspot/os_cpu/bsd_zero/atomic_bsd_zero.hpp

```diff
@@ -285,12 +285,31 @@ inline T Atomic::PlatformCmpxchg<8>::operator()(T volatile* dest,
   return value;
 }
 
+// Atomically copy 64 bits of data
+static void atomic_copy64(const volatile void *src, volatile void *dst) {
+#if defined(PPC32)
+  double tmp;
+  asm volatile ("lfd  %0, 0(%1)\n"
+                "stfd %0, 0(%2)\n"
+                : "=f"(tmp)
+                : "b"(src), "b"(dst));
+#elif defined(S390) && !defined(_LP64)
+  double tmp;
+  asm volatile ("ld  %0, 0(%1)\n"
+                "std %0, 0(%2)\n"
+                : "=r"(tmp)
+                : "a"(src), "a"(dst));
+#else
+  *(jlong *) dst = *(const jlong *) src;
+#endif
+}
+
 template<>
 template<typename T>
 inline T Atomic::PlatformLoad<8>::operator()(T const volatile* src) const {
   STATIC_ASSERT(8 == sizeof(T));
   volatile int64_t dest;
-  os::atomic_copy64(reinterpret_cast<const volatile int64_t*>(src), reinterpret_cast<volatile int64_t*>(&dest));
+  atomic_copy64(reinterpret_cast<const volatile int64_t*>(src), reinterpret_cast<volatile int64_t*>(&dest));
```

Comment:
[pre-existing] Why do we have these casts when `atomic_copy64` takes `volatile void*`?  (There are more like this that I didn't bother to comment directly.)

Reply:
To be honest I don't know :-)